docs/tutorials/constraints.rst

```diff
@@ -0,0 +1,255 @@
+  an array of times to test, an `astroplan.Observer` object, and one or a list
+  of `astroplan.FixedTarget` objects. `compute_constraints` will return a matrix
+  of booleans that describe whether or not each target meets the constraints.
+  The super class `Constraint` has a `__call__` method which will run your
+  custom class's `compute_constraints` method when you check if a target is
+  observable using `astroplan.is_observable` or
+  `astroplan.is_always_observable`.
+
+Here's our `VegaSeparationConstraint` implementation::
+
+    from astroplan import Constraint, is_observable
+    from astropy.coordinates import Angle
+
+    class VegaSeparationConstraint(Constraint):
+        """
+        Constraint the separation from Vega
+        """
+        def __init__(self, min=None, max=None):
+            """
+            min : `~astropy.units.Quantity` or `None` (optional)
+                Minimum acceptable separation between Vega and target. `None`
+                indicates no limit.
+            max : `~astropy.units.Quantity` or `None` (optional)
+                Minimum acceptable separation between Vega and target. `None`
+                indicates no limit.
+            """
+            self.min = min
+            self.max = max
+
+        def compute_constraint(self, times, observer, targets):
+
+            # Vega's position is essentially unchanging, but if it were to be a
+            # moving target, we would need an array of coordinates for Vega as a
+            # function of time. Here we'll simulate that behavior with multiple
+            # copies of the Vega coordinate
+            vega = SkyCoord(ra=279.23473479*u.deg, dec=38.78368896*u.deg)
+            vega = SkyCoord(len(times)*[vega])
+
+            # If `targets` is a FixedTarget object, get the SkyCoord
+            target_coords = SkyCoord([target.coord if hasattr(target, 'coord')
+                                      else target for target in targets])
+
+            # Calculate separation between target and vega
+            vega_separation = Angle([vega.separation(target)
+                                     for target in target_coords])
+
+            # If a maximum is specified but no minimum
+            if self.min is None and self.max is not None:
+                mask = vega_separation < self.max
+
+            # If a minimum is specified but no maximum
+            elif self.max is None and self.min is not None:
+                mask = self.min < vega_separation
+
+            # If both a minimum and a maximum are specified
+            elif self.min is not None and self.max is not None:
+                mask = ((self.min < vega_separation) & (vega_separation < self.max))
+
+            # Otherwise, raise an error
+            else:
+                raise ValueError("No max and/or min specified in "
+                                 "VegaSeparationConstraint.")
+
+            # Return the boolean mask
```

Comment:
maybe add just a bit more here along the lines of "return an array that is True where the target is observable and False where it is not"?  The typical user here might not quite know that's what you mean by "mask".